FFmpegInterop/Metadata.h

```diff
@@ -18,8 +18,70 @@
 
 #pragma once
 
+namespace winrt::FFmpegInterop
+{
+	struct FFmpegInteropMSSConfig;
+}
+
 namespace winrt::FFmpegInterop::implementation
 {
-	void PopulateMSSMetadata(_In_ const Windows::Media::Core::MediaStreamSource& mss, _In_ const AVDictionary* metadata);
-	void SetMSSThumbnail(_In_ const Windows::Media::Core::MediaStreamSource& mss, _In_ const AVStream* thumbnailStream);
+	struct PropertyKey
+	{
+		PropertyKey(_In_ PROPERTYKEY key, _In_ VARTYPE type) :
+			key(key),
+			type(type)
+		{
+
+		}
+
+		PROPERTYKEY key;
+		VARTYPE type;
+	};
+
+	inline const std::unordered_map<std::string, PropertyKey> c_metadataPropertiesMap
+	{

```

Comment:
Where does this list come from?

Reply:
The keys come from:
- [metadata_api · FFmpeg/libavformat/avformat.h · FFmpeg/FFmpeg (github.com)](https://github.com/search?q=repo%3AFFmpeg%2FFFmpeg%20metadata_api&type=code)
  - FFmpeg's list of generic tag names
- [Ogg Vorbis Documentation (xiph.org)](https://xiph.org/vorbis/doc/v-comment.html)
  - List of standard tags for Ogg files
- [Tags (matroska.org)](https://www.matroska.org/technical/tagging.html)
  - List of official tags for Matroska files

I referenced internal MF media sources, [Metadata Properties for Media Files - Win32 apps | Microsoft Learn](https://learn.microsoft.com/en-us/windows/win32/medfound/metadata-properties-for-media-files), propkey.h from the Windows SDK, Shell's internal mediaprops.xml, and used my best judgement to come up with the mappings.